libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2073,6 +2074,20 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 					errorMsg += " Use \"address(" + varName + ")." + memberName + "\" to access this address member.";
 					break;
 				}
+		}
+		else if (exprType->category() == Type::Category::Address)
+		{
+			// Trigger error when using send or transfer with a non-payable fallback function.
+			auto addressType = dynamic_cast<AddressType const*>(exprType.get());
+
+			if (
+				(memberName == "send" || memberName == "transfer") &&
+				addressType->stateMutability() != StateMutability::Payable
```

Comment:
Actually we should be able to move this part of the condition (mutability != Payable) into the ``if``-body as an assertion, right? If the address is in fact payable the members should have been found and we should never end up here anyways...